File: solutions/javascript_nodejs/03.cafe-bot/dialogs/shared/stateProperties/entityProperty.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

module.exports = {
    EntityProperty: class {
        /**
         * Entity Property constructor.
         *
         * @param {String} name entity name
         * @param {String} value entity value
         */
        constructor(name, value) {
            if (!name || !value) throw new Error('Need name and value to create an entity');
            this.entityName = name;
            this.entityValue = value;
        }
    }
};

File: samples/csharp_dotnetcore/51.cafe-bot/Dialogs/Shared/StateProperties/EntityProperty.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

namespace Microsoft.BotBuilderSamples
{
    public class EntityProperty
    {
        public EntityProperty(string name, object value)
        {
            EntityName = name;
            Value = value;
        }

        public string EntityName { get; }

        public object Value { get; }
    }
}
